src/main/webapp/app/exercises/shared/exercise/exercise.service.ts

```diff
@@ -56,12 +56,21 @@ export class ExerciseService {
     validateDate(exercise: Exercise) {
         exercise.dueDateError = exercise.releaseDate && exercise.dueDate ? !exercise.dueDate.isAfter(exercise.releaseDate) : false;
 
-        exercise.assessmentDueDateError =
-            exercise.assessmentDueDate && exercise.releaseDate
-                ? !exercise.assessmentDueDate.isAfter(exercise.releaseDate)
-                : exercise.assessmentDueDate && exercise.dueDate
-                ? !exercise.assessmentDueDate.isAfter(exercise.dueDate)
-                : false;
+        if (exercise.releaseDate && exercise.assessmentDueDate) {
+            if (exercise.dueDate) {
+                exercise.assessmentDueDateError = !exercise.assessmentDueDate.isAfter(exercise.dueDate) && !exercise.assessmentDueDate.isAfter(exercise.releaseDate);
+            } else {
+                exercise.assessmentDueDateError = !exercise.assessmentDueDate.isAfter(exercise.releaseDate);
+            }
+        }
+
+        if (exercise.dueDate && exercise.assessmentDueDate) {
```

Comment:
This check is already done if `releaseDate` is set?

```suggestion
        } else if (exercise.dueDate && exercise.assessmentDueDate) {
```